py3status/modules/uptime.py

```diff
@@ -63,63 +62,29 @@ class Py3status:
     format = 'up {days} days {hours} hours {minutes} minutes'
 
     def post_config_hook(self):
-        self._decades = self.py3.format_contains(self.format, 'decades')
-        self._years = self.py3.format_contains(self.format, 'years')
-        self._weeks = self.py3.format_contains(self.format, 'weeks')
-        self._days = self.py3.format_contains(self.format, 'days')
-        self._hours = self.py3.format_contains(self.format, 'hours')
-        self._minutes = self.py3.format_contains(self.format, 'minutes')
-        self._seconds = self.py3.format_contains(self.format, 'seconds')
+        self.dict = {}
+        t = ['decades', 'years', 'weeks', 'days', 'hours', 'minutes', 'seconds']
+        s = [315360000, 31536000, 604800, 86400, 3600, 60, 1]
+        for unit, second in zip(t, s):
+            self.dict[unit] = {
+                'start': self.py3.format_contains(self.format, unit),
+                'seconds': second
+            }
 
     def uptime(self):
-        # Units will be computed from bare seconds since timedelta only
-        # provides .days and .seconds anyway. Getting rid of the seconds
-        # part. Keeping the floating point part would make divmod return
-        # floats, and thus would require days/hours/minutes/seconds to be
-        # casted to int before formatting, which would be dirty to handle
-        # since we can't cast None to int.
         with open('/proc/uptime', 'r') as f:
-            up = int(float(f.readline().split()[0]))
-            offset = time() - up
-
-        cache_timeout = decades = years = weeks = days = hours = minutes = seconds = 0
-
-        # Decades
-        if self._decades:
-            decades, up = divmod(up, 315360000)  # 10 years -> decade
-            cache_timeout = 315360000
-        # Years
-        if self._years:
-            years, up = divmod(up, 31536000)  # 365 days -> year
-            cache_timeout = 31536000
-        # Weeks
-        if self._weeks:
-            weeks, up = divmod(up, 604800)  # 7 days -> week
-            cache_timeout = 604800
-        # Days
-        if self._days:
-            days, up = divmod(up, 86400)  # 24 hours -> day
-            cache_timeout = 86400
-        # Hours
-        if self._hours:
-            hours, up = divmod(up, 3600)  # 60 minutes -> hour
-            cache_timeout = 3600
-        # Minutes
-        if self._minutes:
-            minutes, up = divmod(up, 60)  # 60 seconds -> minute
-            cache_timeout = 60
-        # Seconds
-        if self._seconds:
-            seconds = up  # 1000000000 nanoseconds -> second
-            cache_timeout = 1
-
-        uptime = self.py3.safe_format(self.format, dict(
-            decades=decades, years=years, weeks=weeks, days=days,
-            hours=hours, minutes=minutes, seconds=seconds))
+            uptime = {'up': int(float(f.readline().split()[0]))}
```

Comment:
it would be better to just do `up = int(float(f.readline().split()[0])` here.

Reply:
We make fresh dict `uptime = {'up': up}` on loops here.

Reply:
yeah but `up` there works too.  removes the `uptime['up']`

Reply:
Make a diff? I think this would create lines as I need to initialize a `uptime` dict first before doing anything with it.... so I just put it in the dict too.

Reply:
yeah it makes a diff.  The code becomes easier to read and we don't have something in the uptime dict that isn't needed :)